front/pages/api/w/[wId]/data_sources/request-email.ts

```diff
@@ -0,0 +1,78 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import sanitizeHtml from "sanitize-html";
+
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { sendEmail } from "@app/lib/email";
+import { apiError } from "@app/logger/withlogging";
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+  auth: Authenticator
+) {
+  const owner = auth.getNonNullableWorkspace();
```

Comment:
We can't use nonNullableWorkspace because it throws an error. Here we want to 404 if workspace is null